riak_test/yz_solrq_test.erl

```diff
@@ -124,21 +163,348 @@ confirm_requeue_undelivered([Node|_] = Cluster, PBConn, BKey, Index) ->
+    Index1BKey1 = lists:nth(1, Index1BKeys),
+    Index1BKey2 = lists:nth(2, Index1BKeys),
+    Index1BKey3 = lists:nth(3, Index1BKeys),
+    Index2BKey1 = lists:nth(1, Index2BKeys),
+    Index2BKey2 = lists:nth(2, Index2BKeys),
+    try
+        yz_rt:load_intercept_code(Node),
+        intercept_index_batch(Node, index_batch_throw_exception),
+        %%
+        %% Send two messages through each indexq on the solrq, which
+        %% will trip the fuse on both; however
+        %% because fuse blown events are handled asynchronously,
+        %% we need to sleep enough to allow the solrqs to get notified.
+        %%
+        [Index1BKey1] = put_bkey_objects(PBConn, [Index1BKey1]),
+        [Index1BKey2] = put_bkey_objects(PBConn, [Index1BKey2]),
+        [Index2BKey1] = put_bkey_objects(PBConn, [Index2BKey1]),
+        [Index2BKey2] = put_bkey_objects(PBConn, [Index2BKey2]),
+        timer:sleep(500), % wait for notification
+        %%
+        %% At this point, the two indexqs in yz_solrq_001 corresponding
+        %% to Index1 and Index2, respectively, should be blown.
+        %% Send one more message through one of the Indexqs, which
+        %% will trigger a purge.
+        %%
+        F = fun() ->
+            [Index1BKey3] = put_bkey_objects(PBConn, [Index1BKey3])
+        end,
+        case PurgeStrategy of
+            ?PURGE_NONE ->
+                spawn(F);
+            _ ->
+                F()
+        end
+    after
+        %%
+        %% Revert the intercept, and drain, giving time for the
+        %% fuse to reset.  Commit to Solr so that we can run a query.
+        %%
+        intercept_index_batch(Node, index_batch_call_orig),
+        timer:sleep(?MELT_RESET_REFRESH + 1000), %% wait for fuse reset
+        drain_solrqs(Node),
+        yz_rt:commit(Cluster, Index1),
+        yz_rt:commit(Cluster, Index2)
+    end,
+    %%
+    %% Return the search results for Index1 and Index2.
+    %% The first list is the set of bkeys we wrote for each index.
+    %% The second list is the set that are available for search.
+    %%
+    Index1SearchBKeys = search_bkeys(PBConn, Index1),
+    Index2SearchBKeys = search_bkeys(PBConn, Index2),
+    {
+        {[Index1BKey1, Index1BKey2, Index1BKey3], Index1SearchBKeys},
+        {[Index2BKey1, Index2BKey2],              Index2SearchBKeys}
+    }.
+
+-spec search_bkeys(pid(), index_name()) -> [bkey()].
+search_bkeys(PBConn, Index) ->
+    {ok, {search_results, SearchResults, _Score, _NumFound}} =
+        riakc_pb_socket:search(PBConn, Index, <<"*:*">>),
+    lists:map(
+        fun({_Index, Fields}) ->
+            {
+                {
+                    proplists:get_value(<<"_yz_rt">>, Fields),
+                    proplists:get_value(<<"_yz_rb">>, Fields)
+                },
+                proplists:get_value(<<"_yz_rk">>, Fields)
+            }
+        end,
+        SearchResults
+    ).
+
+%% TODO Revisit re-using Indices and Buckets
```

Comment:
Can probably move the TODO to where we list out the Bucket macros and mention that we should eventually reuse indexes in this test and change the `delete_mode`.